### ISSUE TITLE
Thanks for the YOLO-World model integrated by your team, I wondered how to set the NMS Threshold and the Score Threshold when inference? 

### ISSUE THREAD

**Issue by FantasticZihao** [ultralytics/ultralytics#17264]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

Thanks for the YOLO-World model integrated by your team, I wondered how to set the NMS Threshold and the Score Threshold when inference?

**Comment by UltralyticsAssistant** [comment]
👋 Hello @FantasticZihao, thank you for reaching out and showing interest in the YOLO-World model from Ultralytics 🚀! This is an automated response, but rest assured, an Ultralytics engineer will assist you soon.

We recommend visiting the [Ultralytics Docs](https://docs.ultralytics.com) for both new and experienced users. Our [Python Usage Guide](https://docs.ultralytics.com/usage/python/) includes various examples that might help you configure inference parameters, including NMS and Score Thresholds.

If your question relates to model inference settings, please specify which version of YOLO you are using and provide any relevant code snippets or settings. This information can assist us in providing you with more targeted guidance.

Join the vibrant Ultralytics community that suits you best:
- 🗨️ For real-time support and chat, visit [Discord](https://ultralytics.com/discord).
- 🧠 For in-depth discussions, explore [Discourse](https://community.ultralytics.com).
- 🎩 Share and gather insights on our [Subreddit](https://reddit.com/r/ultralytics).

## Upgrade

Ensure you're using the latest `ultralytics` package. Upgrade as needed in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/):

```bash
pip install -U ultralytics
```

## Environments

For ease of use, consider running YOLO in one of our verified environments. Each setup includes pre-installed dependencies:

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM: [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI: [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**: [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

Verify the operational status of our models:
<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule) tests are passing, ensuring the correct operation of all YOLO modes and tasks on various platforms.

**Comment by FantasticZihao** [comment]
> It uses the same args as normal YOLO.
> 
> https://docs.ultralytics.com/modes/predict/#inference-arguments

Thanks a lot

**Comment by FantasticZihao** [comment]
> It uses the same args as normal YOLO.
> 
> https://docs.ultralytics.com/modes/predict/#inference-arguments

By the way, how many pretrained models can i load?

**Comment by glenn-jocher** [comment]
@FantasticZihao YOLO-World pretrained models that are available are shown at https://docs.ultralytics.com/models/yolo-world/

<img width="1982" alt="Screenshot 2024-10-30 at 18 22 08" src="https://github.com/user-attachments/assets/06c48954-851f-44ed-b796-3da2879b9014">

**Comment by FantasticZihao** [comment]
> @FantasticZihao YOLO-World pretrained models that are available are shown at https://docs.ultralytics.com/models/yolo-world/
> 
> <img alt="Screenshot 2024-10-30 at 18 22 08" width="1982" src="https://private-user-images.githubusercontent.com/26833433/381663602-06c48954-851f-44ed-b796-3da2879b9014.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzAzMzc1NDQsIm5iZiI6MTczMDMzNzI0NCwicGF0aCI6Ii8yNjgzMzQzMy8zODE2NjM2MDItMDZjNDg5NTQtODUxZi00NGVkLWI3OTYtM2RhMjg3OWI5MDE0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMzElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDMxVDAxMTQwNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWJkNmY1MjBmNjU1ZTBkM2MxZGQ1NzY1NDVlZGUyZDA5OTlmODQwMDM3Njg0NzU0YmQ0YTM5ZWM4YTZjY2NhN2MmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.5MFmI5Ba6eOzTIoGjBShwq8nJyqsz-Jn3Q_zDHmA_cI">

Thanks for your reply, i wondered if the weights files of ultralytics  are the same as those of author's open source? In other words, the weights files  is whether acquired by pretraining on O365+GoldG+CC3M-Lite datasets? Actually, i try the same picture on hugging face demo and ultralytics, but finding the inferior result on ultralytics even  used the largest weights provided by ultralytics.

**Comment by glenn-jocher** [comment]
The weights provided by Ultralytics may differ from those of the original authors, as they could be trained on different datasets or with different configurations. If you're noticing discrepancies in performance, it might be due to these differences in training data or model settings. For optimal results, ensure you're using the correct model and configuration for your specific use case.
